### ISSUE TITLE
ContainsAny operator returns error

### ISSUE THREAD

**Issue by StefanBogdan** [weaviate/weaviate#3559]
If I run this query on Weaviate version `1.21.3`:
```graphql
{
  Get {
    MyClass(
      nearText: {concepts: ["some concept"]}
      where: {
        path: ["path"],
        valueText: ["SomeValue"],
        operator: ContainsAny
      }
    ){
      _additional {
        distance
      }
    }
  }
}
```
It runs and returns this error after around 7 seconds:

```json
{
  "data": {
    "Get": {
      "MyClass": null
    }
  },
  "errors": [
    {
      "locations": [
        {
          "column": 5,
          "line": 3
        }
      ],
      "message": "explorer: get class: vector search: object vector search at index grant: remote shard aqkmTCSWwqmw: status code: 500, error: shard grant_aqkmTCSWwqmw: build inverted filter allow list: value type should be []string but is []interface {}\n: context deadline exceeded",
      "path": [
        "Get",
        "MyClass"
      ]
    }
  ]
}
```

If I run it with `Equal` operator it works:
```graphql
{
  Get {
    MyClass(
      nearText: {concepts: ["some concept"]}
      where: {
        path: ["path"],
        valueText: "SomeValue",
        operator: Equal
      }
    ){
      _additional {
        distance
      }
    }
  }
}
```

Seems like a bug in `ContainsAny`.

**Comment by felixng313** [comment]
+1

**Comment by felixng313** [comment]
The issue is occurring in a cluster where the data is stored on a remote node.

**Comment by etiennedi** [comment]
Thank you @felixng313. The team had looked into this originally when @StefanBogdan raised the issue, but hadn't been able to reproduce it yet. Do you have any steps to reproduce or possibly a small data dump or backup, we could use for investigation? Thank you.

**Comment by felixng313** [comment]
docker-compose.yml
```
version: '3.4'
services:
  weaviate-node-1:
    image: semitechnologies/weaviate:1.22.0
    ports:
    - 8880:8080
    restart: on-failure:0
    environment:
      QUERY_DEFAULTS_LIMIT: 25
      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
      DEFAULT_VECTORIZER_MODULE: 'none'
      CLUSTER_HOSTNAME: 'node1'
      CLUSTER_GOSSIP_BIND_PORT: '7100'
      CLUSTER_DATA_BIND_PORT: '7101'
      
  weaviate-node-2:
    image: semitechnologies/weaviate:1.22.0
    ports:
    - 8881:8080
    restart: on-failure:0
    environment:
      QUERY_DEFAULTS_LIMIT: 25
      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
      DEFAULT_VECTORIZER_MODULE: 'none'
      CLUSTER_HOSTNAME: 'node2'
      CLUSTER_GOSSIP_BIND_PORT: '7102'
      CLUSTER_DATA_BIND_PORT: '7103'
      CLUSTER_JOIN: 'weaviate-node-1:7100'  # This must be the service name of the "founding" member node.
```

Create a simple schema without any data
```
curl --location 'http://localhost:8880/v1/schema' \
--header 'Content-Type: application/json' \
--data '{
        "class": "Article",
        "description": "A written text, for example a news article or blog post",
        "properties": [
            {
            "dataType": [
                "string"
            ],
            "description": "Title of the article",
            "name": "title"
            },
            {
            "dataType": [
                "text"
            ],
            "description": "The content of the article",
            "name": "content"
            }
        ]
    }'

```

And make a search with filter "ContainsAnyt"
```
{
  Get {
    Article (
      bm25: {
        	query: "content"
      }
      where: {
          path: ["content"],
          operator: ContainsAny,
          valueText: ["abc"]
        }
    ) {
      content
      
    }
  }
}
```

Then an exception will occur.

**Comment by etiennedi** [comment]
Thank you so much! We'll check it out right away!

**Comment by antas-marcin** [comment]
PR #3709 fixes this. This fix will be ported to `v1.22` and also to `v1.21` Weaviate version

**Comment by etiennedi** [comment]
Fixed in `v1.21.9` and `v1.22.2`.
